packages/core/content-manager/server/src/history/utils.ts

```diff
@@ -0,0 +1,15 @@
+import { Strapi } from '@strapi/types';
+import { type services } from './services';
+
+type HistoryServiceName = keyof typeof import('./services').services;
+
+type HistoryServices = {
+  [K in HistoryServiceName]: ReturnType<(typeof services)[K]>;
+};
+
+function getService<T extends keyof HistoryServices>(strapi: Strapi, name: T) {
+  // Cast is needed because the return type of strapi.service is too vague
+  return strapi.service(`plugin::content-manager.${name}`) as HistoryServices[T];
+}
```

Comment:
would this typing have the same result?

```suggestion
type HistoryServices =  typeof import('./services').services;

function getService<T extends keyof HistoryServices>(strapi: Strapi, name: T) {
  // Cast is needed because the return type of strapi.service is too vague
  return strapi.service(`plugin::content-manager.${name}`) as HistoryServices[T];
}
```

Reply:
I had to add a ReturnType to your line 8, but yes :)